classes/Twig/Block/ChannelInfoBlock.php

```diff
@@ -79,7 +84,130 @@ public function render()
+            if ($phpVersions === $previousPHPRange) {
+                $previousPrestaVersion = $prestashopVersion;
+            } else {
+                $label = $this->buildPSLabel($startPrestaShopVersion, $previousPrestaVersion);
+                $result[$label]['php_versions'] = $this->buildPhpVersionsList($previousPHPRange);
+                $result[$label]['is_current'] = $isCurrentPrestaVersion;
+                $startPrestaShopVersion = $prestashopVersion;
+                $result[$label]['is_target'] = $this->getFormattedVersion($this->channelInfo->getInfo()['version_num'], self::PS_VERSION_DISPLAY_MAX_PRECISION) === $label;
+                $previousPrestaVersion = null;
+            }
+            if ($i === count(UpgradeSelfCheck::PHP_PS_VERSIONS)) {
+                $result[$prestashopVersion]['php_versions'] = $this->buildPhpVersionsList($phpVersions);
+                $result[$prestashopVersion]['is_current'] = $isCurrentPrestaVersion;
+                $result[$prestashopVersion]['is_target'] = $this->getFormattedVersion($this->channelInfo->getInfo()['version_num'], self::PS_VERSION_DISPLAY_MAX_PRECISION) === $prestashopVersion;
+            }
+            $previousPHPRange = $phpVersions;
+        }
+
+        return $result;
+    }
+
+    /**
+     * Builds PrestaShop version label for display
+     *
+     * @param string $startVersion
+     * @param string $endVersion
+     *
+     * @return string
+     */
+    public function buildPSLabel($startVersion, $endVersion)
+    {
+        if ($startVersion === self::PS_MINIMAL_VERSION) {
+            return '>= ' . self::PS_MINIMAL_VERSION;
+        }
+
+        return $startVersion .= $endVersion ? ' ~ ' . $endVersion : '';
+    }
+
+    /**
+     * Builds a list of php versions for a given php version range
+     *
+     * @param array $phpVersionRange
+     *
+     * @return array
+     */
+    public function buildPhpVersionsList($phpVersionRange)
+    {
+        $phpStart = $phpVersionRange[0];
+        $phpEnd = $phpVersionRange[1];
+        $phpVersionsList = [];
+        $inRange = false;
+        foreach (UpgradeSelfCheck::PHP_VERSIONS_DISPLAY as $phpVersion) {
+            if ($phpVersion === $phpStart) {
+                $inRange = true;
+            }
+            if ($inRange) {
+                $phpVersionsList[] = $phpVersion;
+            }
+            if ($phpVersion === $phpEnd) {
+                break;
+            }
+        }
+
+        return $phpVersionsList;
+    }
+
+    /**
+     * Find out if a given prestashop version is equal to the one currently used
+     * (not taking patch versions into account)
+     *
+     * @param string $prestaversion
+     *
+     * @return bool
+     */
+    public function isCurrentPrestashopVersion($prestaversion)
+    {
+        // special case for 1.6.1 versions
+        if (substr(_PS_VERSION_, 0, 5) === '1.6.1' && $prestaversion === self::PS_MINIMAL_VERSION) {
```

Comment:
What if the version is lower than PS_MINIMAL_VERSION? :thinking:

Reply:
then the function will return false...  (it's unit tested)